src/metabase/api/table.clj

```diff
@@ -51,8 +51,14 @@
 (api/defendpoint GET "/uploaded"
   "Get all `Tables` visible to the current user which were created by uploading a file."
   []
-  (as-> (t2/select Table, :active true, :is_upload true, {:order-by [[:name :asc]]}) tables
-        (filterv mi/can-read? tables)))
+  (->> (t2/select Table {:select    [:t.*, [[:count :c.id] :usage_count]]
+                         :from      [[:metabase_table :t]]
+                         :left-join [[:report_card :c] [:= :t.id :c.table_id]]
```

Comment:
I chatted to @calherries about this simplification, and he convinced me that it's not OK to have "false zeros", i.e. false negatives on whether it's safe to delete a table. 

One idea that he had was to populate this usage data via a mechanism like we have planned for "last used at", where QP execution writes metadata on what it used. 

For now I think it's better to just omit this field than for it to be misleading.